File: Programmers/Sort_01.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int cnt = 0;
        
        for(int idx=0; idx<commands.length; idx++) {
            int arrayLength = commands[idx][1] - commands[idx][0] + 1;
            int[] arr = new int[arrayLength];
            System.arraycopy(array, commands[idx][0]-1, arr, 0, arrayLength);
            Arrays.sort(arr);
            answer[cnt] = arr[commands[idx][2] - 1];
            cnt++;
        }
        
        
        return answer;
    }
}

File: Programmers/Sort_01.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = IntArray(commands.size)
        
        for(i in 0..answer.size - 1) {
            var arr = array.slice(commands[i][0] - 1..commands[i][1] - 1)
            arr = arr.sorted()
            answer[i] = arr[commands[i][2] - 1]
        }
        
        return answer
    }
    
    fun solution2(array: IntArray, commands: Array<IntArray>): IntArray {
        return commands.map {
            array.slice(IntRange(it[0] - 1, it[1] - 1)).sorted()[it[2] - 1]
        }.toIntArray()
    }
}
